source/common/stats/hystrix.cc

```diff
@@ -43,61 +42,97 @@ uint64_t Hystrix::getRollingValue(std::string cluster_name, std::string stats) {
   }
 }
 
-void Hystrix::updateRollingWindowMap(Stats::Store& stats, std::string cluster_name) {
-  std::string prefix = "cluster." + cluster_name + ".";
-
-  // combining timeouts+retries - retries are counted  as separate requests
-  // (alternative: each request including the retries counted as 1)
-  uint64_t timeouts = stats.counter(prefix + "upstream_rq_timeout").value() +
-                      stats.counter(prefix + "upstream_rq_per_try_timeout").value();
+void Hystrix::CreateCounterNameLookupForCluster(const std::string& cluster_name) {
+	// Building lookup name map for all specific cluster values.
+	// Every call to the updateRollingWindowMap function should get the appropriate name from the map.
+	std::string cluster_name_with_prefix = absl::StrCat("cluster.", cluster_name, ".");
+	counter_name_lookup[cluster_name]["upstream_rq_timeout"] =
+		absl::StrCat(cluster_name_with_prefix, "upstream_rq_timeout");
+	counter_name_lookup[cluster_name]["upstream_rq_per_try_timeout"] =
+		absl::StrCat(cluster_name_with_prefix, "upstream_rq_per_try_timeout");
+	counter_name_lookup[cluster_name]["timeouts"] =
+		absl::StrCat(cluster_name_with_prefix, "timeouts");
+	counter_name_lookup[cluster_name]["upstream_rq_5xx"] =
+		absl::StrCat(cluster_name_with_prefix, "upstream_rq_5xx");
+	counter_name_lookup[cluster_name]["retry.upstream_rq_5xx"] =
+		absl::StrCat(cluster_name_with_prefix, "retry.upstream_rq_5xx");
+	counter_name_lookup[cluster_name]["upstream_rq_4xx"] =
+		absl::StrCat(cluster_name_with_prefix, "upstream_rq_4xx");
+	counter_name_lookup[cluster_name]["retry.upstream_rq_4xx"] =
+		absl::StrCat(cluster_name_with_prefix, "retry.upstream_rq_4xx");
+	counter_name_lookup[cluster_name]["errors"] =
+		absl::StrCat(cluster_name_with_prefix, "errors");
+	counter_name_lookup[cluster_name]["upstream_rq_2xx"] =
+		absl::StrCat(cluster_name_with_prefix, "upstream_rq_2xx");
+	counter_name_lookup[cluster_name]["success"] =
+		absl::StrCat(cluster_name_with_prefix, "success");
+	counter_name_lookup[cluster_name]["upstream_rq_pending_overflow"] =
+		absl::StrCat(cluster_name_with_prefix, "upstream_rq_pending_overflow");
+	counter_name_lookup[cluster_name]["rejected"] =
+		absl::StrCat(cluster_name_with_prefix, "rejected");
+	counter_name_lookup[cluster_name]["total"] =
+		absl::StrCat(cluster_name_with_prefix, "total");
+}
 
-  pushNewValue(prefix + "timeouts", timeouts);
+void Hystrix::updateRollingWindowMap(Stats::Store& stats, const std::string& cluster_name) {
+  if (counter_name_lookup.find(cluster_name)==counter_name_lookup.end()) {
+    CreateCounterNameLookupForCluster(cluster_name);
+  }
```

Comment:
I guess here the idea was to avoid repeating this action every time? If it is called from updateRollingWindowMap() this code will still be called every time.